cmd/agent/agent.go

```diff
@@ -0,0 +1,126 @@
+package main
+
+import (
+	"fmt"
+	"net/http"
+	"os"
+
+	// Adds version information
+	_ "github.com/grafana/agent/pkg/build"
+
+	"github.com/grafana/agent/pkg/integrations"
+	"github.com/grafana/agent/pkg/loki"
+	"github.com/grafana/agent/pkg/tempo"
+
+	"github.com/go-kit/kit/log/level"
+	"github.com/grafana/agent/pkg/config"
+	"github.com/grafana/agent/pkg/prom"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/weaveworks/common/server"
+
+	// Register Prometheus SD components
+	_ "github.com/prometheus/prometheus/discovery/install"
+
+	// Register integrations
+	_ "github.com/grafana/agent/pkg/integrations/install"
+
+	"github.com/go-kit/kit/log"
+)
+
+type AgentServer struct {
```

Comment:
Hmm, AgentServer doesn't necessarily feel like the right name here. This should probably just be `Agent` I think.

Reply:
I originally had Agent here, but wanted it to match the struct name. I can make them both but worried about overusing the term Agent, that being said also throwing the word server around.

Reply:
We can pick another generic name, like Entrypoint, too